src/main/java/zd/zero/waifu/motivator/plugin/actions/MotivateMeAction.java

```diff
@@ -28,24 +29,25 @@ public void actionPerformed( @NotNull AnActionEvent e ) {
             pluginState.isMotivateMeEnabled(),
             pluginState.isMotivateMeSoundEnabled() );
 
-        doOrElse(
-            VisualMotivationAssetProvider.INSTANCE.pickAssetFromCategories(
-                WaifuAssetCategory.CELEBRATION,
-                WaifuAssetCategory.HAPPY,
-                WaifuAssetCategory.SMUG
-            ),
-            motivationAsset ->
-                VisualMotivationFactory.INSTANCE.constructNonTitledMotivation(
-                    Objects.requireNonNull( e.getProject() ),
-                    motivationAsset,
-                    config
-                ).motivate(),
-            () ->
-                UpdateNotification.INSTANCE.sendMessage(
-                    "'Motivate Me' Unavailable Offline",
-                    ProjectConstants.getWAIFU_UNAVAILABLE_MESSAGE(),
-                    e.getProject()
-                ) );
+        ApplicationManager.getApplication().executeOnPooledThread( () ->
```

Comment:
How come this is executed on a separate thread?

Reply:
<img width="600" alt="Screen Shot 2020-09-26 at 08 27 49" src="https://user-images.githubusercontent.com/21978370/94325676-2d830e00-ffd2-11ea-8b49-8a503e768033.png">

Double-checked the documentation, and to also confirm that it's running on a separate thread. I added a `Thread#sleep` inside it, and a statement after that block of code and it executes immediately.

```java
        ApplicationManager.getApplication().executeOnPooledThread( () -> {
            try {
                Thread.sleep( 10_000 );
            } catch ( InterruptedException interruptedException ) {
                interruptedException.printStackTrace();
            }
        } );

        System.out.println( "What is life?" );
```

Reply:
I should have been more clear, sorry!

Why does the motivation need to be executed on a separate thread? Asset downloading happens on a separate thread now (and is hidden away in the motivation creation API), the only time it blocks is if it the category hasn't been downloaded yet. So the motivation can be created and invoked  on the AWT thread, with minimal risk of freezing.

It seems a bit unnecessary and can be removed, unless I missed something. 

Just as a FYI these are also other places that motivation is invoked:
- https://github.com/zd-zero/waifu-motivator-plugin/blob/12e07bd274f2c56c0e347473afea4093b122f8d9/src/main/java/zd/zero/waifu/motivator/plugin/personality/core/TaskPersonalityCore.kt#L19
- https://github.com/zd-zero/waifu-motivator-plugin/blob/12e07bd274f2c56c0e347473afea4093b122f8d9/src/main/java/zd/zero/waifu/motivator/plugin/personality/core/IdlePersonalityCore.kt#L27
- https://github.com/zd-zero/waifu-motivator-plugin/blob/12e07bd274f2c56c0e347473afea4093b122f8d9/src/main/java/zd/zero/waifu/motivator/plugin/WaifuMotivatorProject.java#L144

Reply:
> Why does the motivation need to be executed on a separate thread?

Yes, this only happens whenever there's no asset for that category for the first time, as it fallbacks to the original impl to download and use it (this is a blocking process)

https://github.com/zd-zero/waifu-motivator-plugin/blob/a03ad55088043127cb7e1427a9daccd294351b69/src/main/java/zd/zero/waifu/motivator/plugin/assets/AssetManager.kt#L88

There are 2 options we can take, we can make other visual motivation creation in a separate thread or remove this as it happens on the first time the category asset is not available. But I prefer to take the former and made other visual notification as non-blocking all the time. wdyt?

Reply:
Yeah, I think that should be fine. I would like to eventually re-think the Motivation API to abstract the need to execute on a separate thread. Rather than peppering it wherever the motivations are created.

Reply:
> I would like to eventually re-think the Motivation API to abstract the need to execute on a separate thread. Rather than peppering it wherever the motivations are created.

Seems that you've already made this via the topic subscription message.

Reply:
It does cover a majority of the motivations, however there are a couple places and possibly more places in the future they could also appear. However, dis good enough for the time being :smile: